engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ScriptAssertValidator.java

```diff
@@ -6,37 +6,61 @@
  */
 package org.hibernate.validator.internal.constraintvalidators.hv;
 
+import static org.hibernate.validator.internal.util.logging.Messages.MESSAGES;
+
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 
 import org.hibernate.validator.constraints.ScriptAssert;
 import org.hibernate.validator.internal.util.Contracts;
-
-import static org.hibernate.validator.internal.util.logging.Messages.MESSAGES;
+import org.hibernate.validator.internal.util.logging.Log;
+import org.hibernate.validator.internal.util.logging.LoggerFactory;
 
 /**
  * Validator for the {@link ScriptAssert} constraint annotation.
  *
  * @author Gunnar Morling
  * @author Hardy Ferentschik
  * @author Kevin Pollet &lt;kevin.pollet@serli.com&gt; (C) 2011 SERLI
+ * @author Marko Bekhta
  */
 public class ScriptAssertValidator implements ConstraintValidator<ScriptAssert, Object> {
 
+	private static final Log log = LoggerFactory.make();
+
 	private String alias;
+	private String reportOn;
+	private String message;
 	private ScriptAssertContext scriptAssertContext;
 
 	@Override
 	public void initialize(ScriptAssert constraintAnnotation) {
 		validateParameters( constraintAnnotation );
 
 		this.alias = constraintAnnotation.alias();
+		this.reportOn = constraintAnnotation.reportOn();
+		this.message = constraintAnnotation.message();
 		this.scriptAssertContext = new ScriptAssertContext( constraintAnnotation.lang(), constraintAnnotation.script() );
 	}
 
 	@Override
 	public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
-		return scriptAssertContext.evaluateScriptAssertExpression( value, alias );
+		boolean validationResult = scriptAssertContext.evaluateScriptAssertExpression( value, alias );
+
+		if ( !reportOn.isEmpty() ) {
```

Comment:
See my comment above, we shouldn't check for the presence of the field or property getter.